shotgun_api3/shotgun.py

```diff
@@ -675,6 +688,32 @@ def _construct_read_parameters(self,
         params["paging"] = { "entities_per_page": self.config.records_per_page,
                              "current_page": 1 }
 
+        if additional_server_presets is not None:
+            if not isinstance(additional_server_presets, list):
+                msg = "'additional_server_presets' parameter must be a list or None"
+                raise ValueError(msg)
+
+            preset_list = []
+            for preset in additional_server_presets:
+                if not isinstance(preset, list):
```

Comment:
Actually, might be even better to break out this resolve logic into a separate method? Then the above suggested comment could just become the docstring.